packages/perseus-score/src/util/get-scoreable-widgets.ts

```diff
@@ -0,0 +1,33 @@
+import {
+    applyDefaultsToWidgets,
+    getWidgetIdsFromContent,
+} from "@khanacademy/perseus-core";
+
+import isWidgetScoreable from "./is-widget-scoreable";
+
+import type {
+    PerseusRenderer,
+    PerseusWidgetsMap,
+} from "@khanacademy/perseus-core";
+
+/**
+ * Returns the upgraded widgets and the IDs of widgets that should be scored.
+ * Filters out widgets not referenced in content and widgets that are static or ungraded.
+ */
+export default function getScoreableWidgets(
+    perseusRenderData: PerseusRenderer,
+): {
+    upgradedWidgets: PerseusWidgetsMap;
+    scoreableWidgetIds: ReadonlyArray<string>;
+} {
+    // There seems to be a chance that PerseusRenderer.widgets might include
+    // widget data for widgets that are not in PerseusRenderer.content,
+    // so this checks that the widgets are being used before scoring them
```

Comment:
We can update this to be more definitive.


```suggestion
    // Because of how the editor saves data that get's published, there 
    // may be widget configurations in `widgets` that aren't actually 
    // used (they don't show up in the `content` string. For this reason 
    // we extract the actual widgets in use from the `content` string and 
    // only score/validate those.
```